dependencies/ih264d/CMakeLists.txt

```
@@ -182,7 +182,7 @@ target_sources(ih264d PRIVATE
 "decoder/arm/ih264d_function_selector_av8.c"
 "decoder/arm/ih264d_function_selector.c"
 )
-target_compile_options(ih264d PRIVATE -DARMV8)
+target_compile_options(ih264d PRIVATE -DARMV8 $<$<COMPILE_LANGUAGE:ASM,Clang>:-Wno-unused-command-line-argument>)
```

I think the issue on aarch64 Linux is that clang doesn't run the C preprocessor on the assembly files.
According to https://developer.arm.com/documentation/100066/0611/assembling-assembly-code/preprocessing-assembly-code you could either replace the assembly code source file suffix from `.s` to `.S` or use the `-x assembler-with-cpp` option. Also, revert the assembly files back to use the `#ifdef` preprocessor directive.
```suggestion
target_compile_options(ih264d PRIVATE -DARMV8 $<$<COMPILE_LANGUAGE:ASM,Clang>:-Wno-unused-command-line-argument>)

if(NOT MSVC)
    set(CMAKE_ASM_FLAGS "${CFLAGS} -x assembler-with-cpp")
endif()
```

---

Yes, this is the right way. closing this PR in favor of @SSimco 's suggestion.